### DISCUSSION TITLE
[question / help]: gpg public key decryption failed: No secret key

### DISCUSSION THREAD

**Discussion by andresvia** [gopasspw/gopass#2471] [Q&A]
### Summary

I'm a long time user of `pass` recently made the switch to `gopass`.

I acknowledge that this is not a `gopass` issue.

I'm trying to get help to debug why I get this error from gpg (gitter, slack, discourse, "stackoverflow" question, etc, could have worked better for a question, but I couldn't find links to those, sorry)

"manual" decrypt works as intended, I think this confirms my GPG setup is correct.

```
gpg <defaultFlags> --decrypt < .password-store/user/somepw.gpg
<decrypted password here>
```

Decryption with `gopass` fails

```
gopass user/somepw
gpg: public key decryption failed: No secret key
gpg: decryption failed: No secret key
x Decryption Failed: exit status 2

Error: failed to retrieve secret "user/somepw": failed to decrypt
```

I checked the `GOPASS_DEBUG_LOG` file, it wasn't helpful. I tried adding more flags to `gpg.go#defaultArgs`, I only got more verbose messages.

You should know I'm not a regular Windows user I copied my `~/.gnupg` and `~/.password-store` from a Mac.

Please help I'm stuck.

### Steps To Reproduce

In a fresh window machine, get `gopass` from source and compile with `go build`, install `gpg4win` with chocolatey.

### Expected behavior

`gopass` decrypts my password with `gpg` on Windows.

Is there any weird behavior on Windows that I'm not aware?

### Environment

- Windows 10
- `gopass` from `go build`
- `gpg` from `choco` `gpg4win` (also tried installer for gpg4win)
- go 1.19.2
- `gpg --version` -> `2.2.29-unknown`

### Additional context

*update*: I just removed the GPG4WIN installation and I have a working setup now. I guess [git for windows](https://git-scm.com/download/win) installation contains a working installation of GPG. `gopass` was executing GPG4WIN binary and `gpg` has been mapped all the time to `/usr/bin/gpg` which explains why my "manual" decryption works, I'm also guessing GPG4WIN doesn't read `~/.gnupg` config from the Git-Bash environment.

**Comment by AnomalRoil** [comment]
I guess we're not using the "right" binary on Windows when there are multiple GPG installations.  
And also, probably that different GPG installation have different interpretations of the "home directory" on Windows.

I think we should look into adding the path to the gpg binary in our `gopass version` output, that could help debug this kind of issues.

Which shell / terminal were you using, out of curiosity?

**Comment by andresvia** [comment]
I used git-bash terminal.

**Comment by skewballfox** [comment]
running into a similar issue I believe. apparently git comes with [gpg bundled](https://superuser.com/questions/1068980/where-is-my-gpgconf-file-on-windows), and these have different locations they check for gpg data. this can be overridden, as mentioned in one of the answers in the linked post

**Comment by nezia1** [comment]
> running into a similar issue I believe. apparently git comes with [gpg bundled](https://superuser.com/questions/1068980/where-is-my-gpgconf-file-on-windows), and these have different locations they check for gpg data. this can be overridden, as mentioned in one of the answers in the linked post

This fixed my issue. Putting the command out there so that it's easier to find:
`git config --global gpg.program "c:/Program Files (x86)/GnuPG/bin/gpg.exe"`

**Comment by RobertoMaurizzi** [comment]
A missing `pinentry` executable for gpg produces the same error... 
plus you should install the configured one OR restart gog-agent after editing the configuration to change which pinentry you want.
